src/client/services/server-api/index.js

```diff
@@ -25,19 +25,25 @@ const ServerAPI = {
   },
 
   querySearch(query){
-    return fetch(`/pc-client/querySearch?${qs.stringify(query)}`, defaultFetchOpts).then(res => res.json());
+    const queryClone=_.assign({},query);
+    if(/(uniprot:\w+|ncbi:[0-9]+|hgnc:\w+)$/.test(queryClone.q)){queryClone.q=queryClone.q.split(':')[1];}
+    return fetch(`/pc-client/querySearch?${qs.stringify(queryClone)}`, defaultFetchOpts).then(res => res.json());
   },
 
   geneQuery(query){
-    query.genes=query.genes.split(' ');
-    return fetch('/api/validation', {
-      method:'POST',
-      headers: {
-        'Accept': 'application/json',
-        'Content-Type': 'application/json'
-      },
-      body:JSON.stringify(query)
-    }).then(res => res.json()).then(ids=> _.assign(ids,{unrecognized:_.tail(ids.unrecognized)}));//remove padding
+    if(query.genes.length>=1){
+      query.genes= query.genes;
```

Comment:
Necessary?

Reply:
Jeff, what do you mean by 'necessary'?

Reply:
"Removing padding" is related to a previous issue(#767) and has been done in pull request(#782). Should I keep the 'padding' and merge the code later?

Reply:
I'm referring to line 35 `query.genes= query.genes;`. Isn't this assigning a variable reference to itself?